libsolidity/codegen/ir/IRLValue.cpp

```diff
@@ -135,8 +135,20 @@ string IRStorageItem::storeValue(string const& _value, Type const& _sourceType)
 
 string IRStorageItem::setToZero() const
 {
-	solUnimplementedAssert(m_type->isValueType(), "");
-	return storeValue(m_context.utils().zeroValueFunction(*m_type) + "()", *m_type);
+	if (m_type->isValueType())
+		return storeValue(m_context.utils().zeroValueFunction(*m_type) + "()", *m_type);
+
+	if (ArrayType const* arrayType = dynamic_cast<decltype(arrayType)>(m_type))
+		if (!arrayType->isDynamicallySized())
+			return m_context.utils().clearStorageRangeFunction(*arrayType->baseType()) +
```

Comment:
That's not enough, for example if the statically-sized array contains a dynamically-sized array.

Reply:
Ah sorry, maybe it is. But then we should maybe rename `clearStorageRangeFunction`

Reply:
Hmm no I think you're right, it's not enough. But I would still merge this and add support for that in the next PR?

Reply:
Does `clearStorageRangeFunction` work with base types that are smaller than 32 bytes?

Reply:
I have the impression that it would be better to use a `clearArray` function as in the old code generator (i.e. pass the array type, not the base type). This can then also be used if you are removing just the last element of an array of arrays.

Reply:
So it wouldn't just clear the whole array but certain ranges within it?

Reply:
If you remove the last element and it is an array, you can use clearArray on that element to clear it.

Reply:
Basically clearArray is just like `clearStorageElement` specialized for arrays.

Reply:
After looking a bit through the code, the dependencies of how those functions work together is a bit tangled:

![image](https://user-images.githubusercontent.com/424752/61645166-6163f480-aca6-11e9-91ab-24eea7e5a878.png)

In the new code we can't use `StorageItem::setToZero` like the old one does because we can't just create a new instance just to call `setToZero` (missing c'tor parameters) so we need to put the functionality of that function into the `YulUtilFunctions`. 

We're missing something like `clearArray` and I'd say the `clearStorangeRangeFunction` here is the (incomplete) equivalent of the old `clearStorageLoop`. 

Something like `clearDynamicArray` is so far completely missing.